lib/vyasa_web/live/media_live/media_bridge.ex

```diff
@@ -226,31 +247,15 @@ defmodule VyasaWeb.MediaLive.MediaBridge do
     }
   end
 
-  @impl true
-  @doc """
-  On receiving a voice_ack, the written and player contexts are now synced.
-  A playback struct is created that represents this synced-state and the client-side hook is triggerred
-  to register the associated events timeline.
-  """
-  def handle_info(
-        {_, :voice_ack,
+  # assigns necessary states if voice is legit and events can be loaded.
+  defp apply_voice_action(
+         %Socket{} = socket,
          %Voice{
            video: video
-         } = voice},
-        socket
-      ) do
-    %Voice{
-      events: voice_events,
-      title: title,
-      file_path: file_path,
-      duration: duration,
-      meta:
-        %{
-          artists: artists,
-          album: album,
-          artwork: artwork
-        } = meta
-    } = loaded_voice = voice |> Medium.load_events()
+         } = voice,
+         ack_val
```

Comment:
ack_val should be transport independent unless transport itself is handling the order logic of some handshake etc.

To avoid overloading the logic here simplifying the message pass here and adding the ack protocol( patternmatching at source and dest) would be ideal

Reply:
![image](https://github.com/user-attachments/assets/97a4d378-622f-40a3-9a79-83cc851536eb)

Some additional context: 

```
Vyasa.PubSub.subscribe("media:session:" <> id)
    Vyasa.PubSub.publish(:init, :media_handshake, "written:session:" <> id)
```

we have two channels one to written_session and the other to media_session though they share the same id

Reply:
i did a more generic thing but that's unnecessary at. the moment. I shouldn't prematurely support it


therefore, I shall: 
1. Drop ack
2. Do a duplicate check by comparing ids within structs